kvbc/test/blockchain_view_test.cpp

```diff
@@ -0,0 +1,244 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#include "gtest/gtest.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <exception>
+#include <iterator>
+
+#include "blockchain_view.h"
+
+namespace {
+
+using namespace concord::kvbc;
+
+using BlockTimestamp = std::uint64_t;
+
+class TimestampedBlockInfo : public BaseBlockInfo {
+ public:
+  static constexpr auto timestampOffset = 100;
+
+  TimestampedBlockInfo(concordUtils::BlockId id) : BaseBlockInfo{id} {}
+
+  static TimestampedBlockInfo getBlockWithTimestamp(BlockTimestamp ts) {
+    TimestampedBlockInfo ret{0};
```

Comment:
Yes, it always sets the the ID to 0 (or could be an arbitrary value), because we need to provide a value of type `TimestampedBlockInfo` for the search based on timestamp and we don't care about the ID value in the `TimestampCompare` lambda. The `getBlockWithTimestamp()` function is just for convenience. However, I agree that having a separate constructor is clearer - adding it.

Additionally, based on your comment, I have added an entry in the `BlockInfo` requirements - the view requires a constructor that takes the block ID only.

Good point!